YAXLib/Deserialization.cs

```diff
@@ -0,0 +1,1541 @@
+        if (elem == null) // no such element is was found
+        {
+            OnExceptionOccurred(new YAXElementMissingException(
+                    serializationLocation, baseElement),
+                !member.MemberType.IsValueType && _ys.UdtWrapper.IsNotAllowedNullObjectSerialization
+                    ? YAXExceptionTypes.Ignore
+                    : member.TreatErrorsAs);
+        }
+        else
+        {
+            var values = elem.Nodes().OfType<XText>().ToArray();
+            if (values.Length <= 0)
+            {
+                // look for an element with the same name AND a yaxlib:realtype attribute
+                var innerElement = XMLUtils.FindElement(baseElement, serializationLocation,
+                    member.Alias.OverrideNsIfEmpty(_ys.TypeNamespace));
+                if (innerElement != null &&
+                    innerElement.Attribute_NamespaceSafe(_ys.Options.Namespace.Uri + _ys.Options.AttributeName.RealType, _ys.DocumentDefaultNamespace) != null)
+                {
+                    deserializedValue = innerElement.Value;
+                    xElementValue = innerElement;
+                }
+                else
+                {
+                    OnExceptionOccurred(
+                        new YAXElementValueMissingException(serializationLocation,
+                            innerElement ?? baseElement),
+                        !member.MemberType.IsValueType && _ys.UdtWrapper.IsNotAllowedNullObjectSerialization
+                            ? YAXExceptionTypes.Ignore
+                            : member.TreatErrorsAs);
+                }
+            }
+            else
+            {
+                deserializedValue = values[0].Value;
+                values[0].Remove();
+            }
+        }
+
+        return deserializedValue;
+    }
+
+    private string DeserializeFromAttribute(XElement baseElement, ref XElement xElementValue,
+        ref XAttribute xAttributeValue, string serializationLocation, MemberWrapper member)
+    {
+        var deserializedValue = string.Empty;
+
+        // find the parent element from its location
+        var attr = XMLUtils.FindAttribute(baseElement, serializationLocation,
+            member.Alias.OverrideNsIfEmpty(_ys.TypeNamespace));
+        if (attr == null) // if the parent element does not exist
+        {
+            // look for an element with the same name AND a yaxlib:realtype attribute
+            var elem = XMLUtils.FindElement(baseElement, serializationLocation,
+                member.Alias.OverrideNsIfEmpty(_ys.TypeNamespace));
+            if (elem != null && elem.Attribute_NamespaceSafe(_ys.Options.Namespace.Uri + _ys.Options.AttributeName.RealType, _ys.DocumentDefaultNamespace) != null)
+            {
+                deserializedValue = elem.Value;
+                xElementValue = elem;
+            }
+            else
+            {
+                OnExceptionOccurred(new YAXAttributeMissingException(
```

Comment:
better use a method instead of duplicating, right
https://github.com/axunonb/YAXLib/blob/1de5321a7a70b9bac206f2d7dcca608be3f45126/YAXLib/Deserialization.cs#L356-L360
and
https://github.com/axunonb/YAXLib/blob/1de5321a7a70b9bac206f2d7dcca608be3f45126/YAXLib/Deserialization.cs#L378-L383